File: Recursion_and_Backtracking_pepcoding_level_1/2.Recursion_on_array/2.display_arr_reverse.js
const display_arr = (arr, i = 0) => {
    if (i === arr.length) return;

    display_arr(arr, i + 1);
    console.log(arr[i]);
};

console.log(display_arr([-5, 34, 342, 34234, 32]));

// E
// n to 0

// F
// n to 1

// F + E
// n to 1, 0

File: Recursion_and_Backtracking_pepcoding_level_1/2.Recursion_on_array/2.display_arr_reverse.rb
def display_arr arr, i=0
    return if i == arr.size

    display_arr arr, i+1
    puts arr[i]
end

display_arr [10,20,30,40,50]

# // E
# // n to 0

# // F
# // n to 1

# // F + E
# // n to 1, 0
